### ISSUE TITLE
Menu config via front matter fails in 0.55

### ISSUE THREAD

**Issue by jzeneto** [gohugoio/hugo#5828]
It seems like "[simple form menu config](https://gohugo.io/content-management/menus/#a-single-menu)" doesn't work anymore in Hugo 0.55:

I have some pages that I need to display in site's menu. They have something like this in their front matters:

```yaml
# ... other stuff
menu: main
weight: 30
```

I've just updated Hugo from 0.54 to 0.55. In 0.54 the menu logic (below) displays links to these pages (as documented in [single menu entries](https://gohugo.io/content-management/menus/#a-single-menu)). But in 0.55, these links don't get printed (I tried to change `.Name` to `.Title` as documented in 0.55 release notes, without success).

Menu template:

```html
{{ range .Site.Menus.main }}
  {{ if .HasChildren }}
    <a>{{ .Name }}</a>
    {{ range .Children }}
      <a href="{{ .URL }}">{{ .Name }}</a>
    {{ end }}
  {{ else }}
    <a href="{{ .URL }}">{{ .Name }}</a>
  {{ end }}
{{ end }}
```

If I change page's front matter to so-called ["advanced" configuration](https://gohugo.io/content-management/menus/#advanced) (below), it works, i. e., links get printed.

```yaml
menu:
  main:
    weight: 30
```

Is this expected? If so, should it be documented?

(I don't use global `menus.toml` because I need custom permalinks for each language)

**Comment by bep** [comment]
Just a quick note:

This:

```
# ... other stuff
menu: main
weight: 30
```

Is and never has been the same as:

```
menu:
  main:
    weight: 30
```

The weight in the first variant is the **page weight**.

**Comment by bep** [comment]
Fixed in 9e9a1f92baf151f8d840d6b5b963945d1410ce25
